lib/smart_proxy_pulp_plugin/disk_usage.rb

```diff
@@ -0,0 +1,52 @@
+module PulpProxy
+  class DiskUsage
+    include ::Proxy::Util
+    SIZE = { :kilobyte => 1_024, :megabyte => 1_048_576, :gigabyte => 1_073_741_824, :terabyte => 1_099_511_627_776 }
+
+    attr_reader :path, :stat, :size
+
+    def initialize(opts ={})
+      raise(::Proxy::Error::ConfigurationError, 'Unable to continue - must provide a path.') if opts[:path].nil?
+      @pathes_hash = path_hash(opts[:path])
+      @path = @pathes_hash.values.join(' ')
+      @size = SIZE[opts[:size]] || SIZE[:gigabyte]
+      @stat = {}
+      find_df
+      get_stat
+    end
+
+    def to_json
+      stat.to_json
+    end
+
+    private
+
+    attr_reader :command_path
+
+    def find_df
+      @command_path = which('df') || raise(::Proxy::Error::ConfigurationError, 'df command was not found unable to retrieve usage information.')
+    end
+
+    def command
+      "LANG=C #{command_path} -B#{size} #{path}"
```

Comment:
This still spawns a shell, as described in http://ruby-doc.org/stdlib-1.9.3/libdoc/open3/rdoc/Open3.html#method-c-popen3 and http://ruby-doc.org/core-2.2.0/Process.html#method-c-spawn - you need to pass the environment in a hash and then the string to binary with *no spaces*, then all arguments in an array. Only then, no shell is spawned.